doc/module-system/module-system.chapter.md

```diff
@@ -1,14 +1,207 @@
 # Module System {#module-system}
 
+::: {.note}
+This chapter is new and not complete yet. For a gentle introduction to the module system, in the context of NixOS, see [Writing NixOS Modules](https://nixos.org/manual/nixos/unstable/index.html#sec-writing-modules) in the NixOS manual.
+:::
+
 ## Introduction {#module-system-introduction}
 
 The module system is a language for handling configuration, implemented as a Nix library.
 
-Compared to plain Nix, it adds documentation, type checking and composition or extensibility.
+It was historically known as the NixOS module system.
 
-::: {.note}
-This chapter is new and not complete yet. For a gentle introduction to the module system, in the context of NixOS, see [Writing NixOS Modules](https://nixos.org/manual/nixos/unstable/index.html#sec-writing-modules) in the NixOS manual.
-:::
+Its purpose is to "extend" the Nix language with
+
+- **dynamic type checking** to find mistakes early on, improving error messages;
+- **documentation**, so that each option has a description, conveniently located in the file that declares it
```

Comment:
It’s the beginning of the introduction, it’s probably fine to just call it a “configuration option” since we’ve already mentioned that modules are for configuration.